src/pages/global-authorization/forms/FieldGlobalForm.js

```diff
@@ -0,0 +1,216 @@
+import { SaleRepository } from 'src/repositories/SaleRepository'
+import FormShell from 'src/components/Shared/FormShell'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { Box, Grid } from '@mui/material'
+import { useInvalidate } from 'src/hooks/resource'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { AccessControlRepository } from 'src/repositories/AccessControlRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { DataSets } from 'src/resources/DataSets'
+import { CommonContext } from 'src/providers/CommonContext'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { AuthContext } from 'src/providers/AuthContext'
+import { getFormattedNumber } from 'src/lib/numberField-helper'
+import { DataGrid } from 'src/components/Shared/DataGrid'
+
+export default function FieldGlobalForm({ labels, maxAccess, resourceId, resourceName, setErrorMessage }) {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { user, setUser } = useContext(AuthContext)
+  const [isLoading, setIsLoading] = useState(false)
+
+  const invalidate = useInvalidate({
+    endpointId: SystemRepository.ModuleClassRES.qry
+  })
+
+  const detailsFormik = useFormik({
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues: {
+      rows: [],
+      resourceId: resourceId,
+      resourceName: resourceName
+    },
+    onSubmit: async obj => {
+
+      const updatedRows = detailsFormik.values.rows
+        .filter(obj => {
+          return obj.accessLevel != null
+        }).map(({ ...rest }) => ({
+          resourceId: resourceId,
+          ...rest
+        }))
+
+      // Create the resultObject
+      const resultObject = {
+        resourceId: resourceId,
+        controls: updatedRows
+      }
+
+      const response = await postRequest({
+        extension: AccessControlRepository.GlobalControlAuthorizationPack.set2,
+        record: JSON.stringify(resultObject)
+      })
+
+      toast.success('Record Edited Successfully')
+
+      invalidate()
+    }
+  })
+
+  const columns = [
+    {
+      component: 'textfield',
+      label: labels.controlId,
+      name: 'controlId',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'textfield',
+      label: labels.controlName,
+      name: 'name',
+      props: {
+        readOnly: true
+      }
+    },
+    {
+      component: 'resourcecombobox',
+      name: 'accessLevelCombo',
+      label: labels.accessLevel,
+      props: {
+        datasetId: DataSets.AU_RESOURCE_CONTROL_ACCESS_LEVEL,
+        valueField: 'key',
+        displayField: 'value',
+        mapping: [ { from: 'key', to: 'accessLevel' } ,
+          { from: 'value', to: 'accessLevelName' }]
+      },
+      async onChange({ row: { update, newRow } }) {
```

Comment:
I don't think we need an onChange here. The datagrid updates everything by itself. We used to use it when we didn't have mapping. It's used now in cash invoice to do numerical calculations whenever a number changes.